thirdPart/3_jsonPlaceHolderTask.js

```diff
@@ -0,0 +1,41 @@
+const baseUrl = 'https://jsonplaceholder.typicode.com';
+
+const getUsers = async () => {
+    try {
+        const res = await fetch(`${baseUrl}/users`);
+        return await res.json();
+    } catch (error) {
+        console.log('Error fetching users:', error);
+        throw error;
+    }
+};
+
+const getPosts = async () => {
+    try {
+        const res = await fetch(`${baseUrl}/posts`);
+        return await res.json();
+    } catch (error) {
+        console.log('Error fetching posts:', error);
+        throw error;
+    }
+};
+
+const addUsersPosts = async () => {
+    try {
+        const [users, posts] = await Promise.all([getUsers(), getPosts()]);
+        const result = users.map(user => {
+            const myPosts = posts.filter(post => post.userId === user.id);
```

Comment:
here is the gap from performance perspective (Big O(N^2)). if the amount of items will be bigger this code will be slow.
The tasks like that generally solved by creating a Map data structure. Create a map of posts, where the `key` is `userId`, and a `value` is the array of posts connected with this user.
And then, you can iterate through the users and get needed data by key, instead of looping though whole posts lists every time